### PR TITLE
feat: add support for anonymous posting [BD-38] [TNL-9224]

### PR DESCRIPTION
**Pull request by xitij2000**
Hooks up the configuration API to allow posting threads anonymously.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @xitij2000! I've created [OSPR-6213](https://openedx.atlassian.net/browse/OSPR-6213) to keep track of it in JIRA.

As a core committer in this repo, you can merge this once the pull request is approved per the [core committer reviewer requirements](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529675973/Open+edX+Core+Committers#Reviews-and-Ownership) and according to the agreement with your edX Champion.

**Comment by natabene** [comment]
@xitij2000 Thank you for your contribution. Is this ready for our review?

**Review by mehaknasir — APPROVED** [review]
Looks good to me

File: src/discussions/posts/post-editor/PostEditor.jsx
Comment (kaustavb12) [inline src/discussions/posts/post-editor/PostEditor.jsx:302]:
```suggestion
                      onBlur={handleBlur}
                      aria-describedby="Post anonymously checkbox"
```

File: src/discussions/posts/post-editor/PostEditor.jsx
Comment (kaustavb12) [inline src/discussions/posts/post-editor/PostEditor.jsx:316]:
```suggestion
                        onBlur={handleBlur}
                        aria-describedby="Post anonymously to peers checkbox"
```

File: src/discussions/data/api.js
Comment (kaustavb12) [inline src/discussions/data/api.js:1]: Could the new `discussions/data` directory be restructured as `discussions/course-config/data` (or something similar) for 2 main reasons:
1. Ease of understanding for future developers
2. Maintaining consistency with other sub-directories in `discussions` directory

**Review by kaustavb12 — CHANGES_REQUESTED** [review]
Added a few suggestions from my side

File: src/discussions/data/api.js
Comment (xitij2000) [reply src/discussions/data/api.js:1]: I see the structure as follows:
- /data for non-discussions related API and data
- /discusions/data for data that's relevant in multiple places
- /discussions/___/data for data specific to a particular component

I've kept this here since it's more global. We don't have any course-config related components or UI. It's used by other UI.

File: src/discussions/posts/post-editor/PostEditor.jsx
Comment (xitij2000) [reply src/discussions/posts/post-editor/PostEditor.jsx:302]: I don't think this is needed. The ``{intl.formatMessage(messages.anonymousPost)}`` we're passing here is set as the label so it's available for accessibility purposes as a label.

**Review by awais-ansari — APPROVED** [review]

**Review by kaustavb12 — APPROVED** [review]
:+1: Approved

- [x] I tested this: (Tested both anonymous and non-anonymous postings)
- [x] I read through the code
- [x] I checked for accessibility issues
- [x] Includes documentation

**Comment by edx-webhook** [comment]
@xitij2000 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/frontend-app-discussions&entry.867055334=https://github.com/edx/frontend-app-discussions/pull/35&entry.1484655318=https://github.com/xitij2000&entry.752974735=2021-11-09+13:23&entry.1917517419=2021-11-16+08:09&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
